tests/ui_tests/app/api.py

```diff
@@ -0,0 +1,82 @@
+import json
+import allure
+import inspect
+import requests
+
+
+class RequestFailedException(Exception):
+    """Request to ADCM API has status code >= 400"""
+
+
+class APIRequester:
+    """Helper to make requests to ADCM API in a browser-like way"""
+
+    def __init__(self, base_url: str, admin_credentials: dict):
+        self.base_url = base_url.rstrip("/")
+        self.credentials = admin_credentials
+        self.login_endpoint = f'{base_url}/api/v1/token/'
+        self.user_create_endpoint = f'{base_url}/api/v1/user/'
+        self.user_delete_ep_template = f'{base_url}/api/v1/user/' + '{}/'
+        self.password_endpoint = f'{base_url}/api/v1/profile/admin/password/'
+
+    def get_authorization_token(self) -> str:
+        """Returns authorization token for "admin" user"""
+        response = requests.post(self.login_endpoint, json=self.credentials)
+        self._check_response(response)
+        return response.json()['token']
+
+    def get_authorization_header(self) -> dict:
+        """Returns "Authorization" header with token for "admin" user"""
+        token = self.get_authorization_token()
+        return {'Authorization': f'Token {token}'}
+
+    @allure.step('Create new user over API')
+    def create_new_user(self, credentials: dict) -> None:
+        """Creates new user"""
+        self._make_authorized_request('post', self.user_create_endpoint, json=credentials)
+
+    @allure.step('Delete user {username} over API')
+    def delete_user(self, username: str) -> None:
+        """Deletes user by name"""
+        self._make_authorized_request('delete', self.user_delete_ep_template.format(username))
+
+    def change_admin_password(self, credentials: dict):
+        """
+        Change admin (default ADCM user) password.
+        Saves new credentials to `self.credentials`.
+        """
+        with allure.step(f'Change admin password to {credentials["password"]}'):
+            self._make_authorized_request('patch', self.password_endpoint, json=credentials)
+            self.credentials = credentials
+
+    def _make_authorized_request(self, method: str, endpoint: str, **kwargs) -> requests.Response:
+        request_function = getattr(requests, method)
+        response = request_function(endpoint, headers=self.get_authorization_header(), **kwargs)
+        self._check_response(response)
+        return response
+
+    @staticmethod
+    def _check_response(response: requests.Response):
+        if (status_code := response.status_code) < 400:
+            return
+        caller_function_name = inspect.stack()[1][3].replace('_', ' ')
+        try:
+            response_content = response.json()
+        except json.JSONDecodeError:
+            response_content = response.text
+        raise RequestFailedException(
+            f'Failed to {caller_function_name} over API. '
+            f'Request finished with status {status_code} '
+            f'and json body: {response_content}',
+        )
```

Comment:
There is no need to write func name in the error since it will still occur inside some allure step

Reply:
Thought it would be nice to add a name to the exception itself, but on other hand, it's not very useful without a report. Removed.